SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -6170,6 +6194,214 @@ void CEulerSolver::SetActDisk_BCThrust(CGeometry *geometry, CSolver **solver_con
+      Marker_Tag = config->GetMarker_All_TagBound(iMarker);
+      ifstream ActDisk_file;
+      /*--- Open the file that contains the propeller data. ---*/
+      ActDisk_file.open(ActDisk_filename.data(), ios::in);
+
+      /*--- Error message if the propeller data input file fails to open. ---*/
+      if (ActDisk_file.fail()) SU2_MPI::Error("Unable to open Actuator Disk Input File", CURRENT_FUNCTION);
+
+        string text_line, text_line_appo, name[2];
+        string::size_type position;
+
+        while (getline (ActDisk_file, text_line)) {
+          /*--- Check if there is the "MARKER_ACTDISK=" string in the current line. If not keep on reading. ---*/
+          position = text_line.find ("MARKER_ACTDISK=");
+          if(position == string::npos){continue;}
+          text_line.erase (0,15);
+          /*--- Read the names of the two faces of the actuator disk and assign them to the name[] array. ---*/
+          istringstream NameID(text_line);
+          for (int i = 0; i < 2; i++){
+            NameID >> name[i];
+          }
+
+          /*--- Check if the propeller data correspond to the actual BC marker. ---*/
+          if (Marker_Tag == name[0] || Marker_Tag == name[1]){
+            /*--- Read and assign the coordinates of the actuator disk center. ---*/
+            getline (ActDisk_file, text_line_appo);
+            text_line_appo.erase (0,7);
+            istringstream C_value(text_line_appo);
+            for (iDim = 0; iDim < nDim; iDim++){
+              C_value >> AD_Center[iDim];
+            }
+
+            /*--- Read and assign the components of the actuator disk axis versor pointing backward. ---*/
+            getline (ActDisk_file, text_line_appo);
+            text_line_appo.erase (0,5);
+            istringstream axis_value(text_line_appo);
+            for (iDim = 0; iDim < nDim; iDim++){
+              axis_value >> AD_Axis[iDim];
+            }
+
+            /*--- Read and assign the value of the actuator disk radius. ---*/
+            getline (ActDisk_file, text_line_appo);
+            text_line_appo.erase (0,7);
+            istringstream R_value(text_line_appo);
+             R_value >> AD_Radius;
+
+             /*--- Read and assign the value of the actuator disk advance ratio. ---*/
+             getline (ActDisk_file, text_line_appo);
+             text_line_appo.erase (0,10);
+             istringstream J_value(text_line_appo);
+             J_value >> AD_J;
+
+             /*--- Read and assign the number of radial stations contained in the propeller data file. ---*/
+             getline (ActDisk_file, text_line_appo);
+             text_line_appo.erase (0,5);
+             istringstream row_value(text_line_appo);
+             row_value >> nRow;
+
+             /*--- Assign the vectors dimension. ---*/
+             rad_v = new su2double [nRow];
+             dCt_v = new su2double [nRow];
+             dCp_v = new su2double [nRow];
+             dCr_v = new su2double [nRow];
```

Comment:
It does sound like the more robust way of doing it, but this code will likely leak memory, because the allocation is done inside the loop and the deallocation outside.
It would be safer (and avoid leaks) to use either std::vector or su2activevector for these variables that need to be resized dynamically.

Reply:
Thank you for the advice, I have changed them as `std::vector`.